yt_dlp/extractor/youtube.py

```diff
@@ -312,250 +452,22 @@ def _real_initialize(self):
-            'INNERTUBE_API_VERSION': 'v1',
-            'INNERTUBE_CLIENT_NAME': 'MWEB',
-            'INNERTUBE_CLIENT_VERSION': '2.20210721.07.00',
-            'INNERTUBE_API_KEY': 'AIzaSyDCU8hByM-4DrUqRUYnGn-3llEO78bcxq8',
-            'INNERTUBE_CONTEXT': {
-                'client': {
-                    'clientName': 'MWEB',
-                    'clientVersion': '2.20210721.07.00',
-                    'clientScreen': 'EMBED',
-                    'hl': 'en',
-                }
-            },
-            'INNERTUBE_CONTEXT_CLIENT_NAME': 2
-        },
-    }
-
-    _YT_DEFAULT_INNERTUBE_HOSTS = {
-        'DIRECT': 'youtubei.googleapis.com',
-        'WEB': 'www.youtube.com',
-        'WEB_REMIX': 'music.youtube.com',
-        'ANDROID_MUSIC': 'music.youtube.com'
-    }
-
-    # clients starting with _ cannot be explicity requested by the user
-    _YT_CLIENTS = {
-        'android': 'ANDROID',
-        'android_music': 'ANDROID_MUSIC',
-        'android_embedded': 'ANDROID_EMBEDDED_PLAYER',
-        'android_agegate': 'ANDROID_AGEGATE',
-        'ios': 'IOS',
-        'ios_music': 'IOS_MUSIC',
-        'ios_embedded': 'IOS_MESSAGES_EXTENSION',
-        'ios_agegate': 'IOS_AGEGATE',
-        'web': 'WEB',
-        'web_music': 'WEB_REMIX',
-        'web_embedded': 'WEB_EMBEDDED_PLAYER',
-        'web_agegate': 'WEB_AGEGATE',
-        'mweb': 'MWEB',
-        'mweb_agegate': 'MWEB_AGEGATE',
-    }
-
-    def _get_default_ytcfg(self, client='WEB'):
-        if client in self._YT_DEFAULT_YTCFGS:
-            return copy.deepcopy(self._YT_DEFAULT_YTCFGS[client])
-        self.write_debug(f'INNERTUBE default client {client} does not exist - falling back to WEB client.')
-        return copy.deepcopy(self._YT_DEFAULT_YTCFGS['WEB'])
+    def _get_default_ytcfg(self, client='web'):
+        return copy.deepcopy(INNERTUBE_CLIENTS[client])
 
-    def _get_innertube_host(self, client='WEB'):
-        return dict_get(self._YT_DEFAULT_INNERTUBE_HOSTS, (client, 'WEB'))
+    def _get_innertube_host(self, client='web'):
+        return INNERTUBE_CLIENTS[client]['INNERTUBE_HOST']
 
-    def _ytcfg_get_safe(self, ytcfg, getter, expected_type=None, default_client='WEB'):
+    def _ytcfg_get_safe(self, ytcfg, getter, expected_type=None, default_client='web'):
         # try_get but with fallback to default ytcfg client values when present
         _func = lambda y: try_get(y, getter, expected_type)
         return _func(ytcfg) or _func(self._get_default_ytcfg(default_client))
 
-    def _extract_client_name(self, ytcfg, default_client='WEB'):
-        return self._ytcfg_get_safe(ytcfg, lambda x: x['INNERTUBE_CLIENT_NAME'], compat_str, default_client)
+    def _extract_client_name(self, ytcfg, default_client='web'):
+        return (
+            try_get(ytcfg, lambda x: x['INNERTUBE_CLIENT_NAME'], compat_str)
+            or self._ytcfg_get_safe(
+                ytcfg, lambda x: x['INNERTUBE_CONTEXT']['client']['clientName'], compat_str, default_client))
```

Comment:
```suggestion
       return self._ytcfg_get_safe(
            ytcfg, (lambda x: x['INNERTUBE_CLIENT_NAME'],
                    lambda x: x['INNERTUBE_CONTEXT']['client']['clientName']), compat_str, default_client)
```
I think would be better?